contracts/tests/bubblegum-test-rpc.ts

```diff
@@ -0,0 +1,512 @@
+import * as anchor from "@project-serum/anchor";
+import {keccak_256} from "js-sha3";
+import {BN, Provider, Program} from "@project-serum/anchor";
+import {Bubblegum} from "../target/types/bubblegum";
+import {Gummyroll} from "../target/types/gummyroll";
+import fetch from "node-fetch";
+import {PROGRAM_ID} from "@metaplex-foundation/mpl-token-metadata";
+import {
+  PublicKey,
+  Keypair,
+  SystemProgram,
+  Transaction,
+  Connection as web3Connection,
+  SYSVAR_RENT_PUBKEY, AccountMeta,
+} from "@solana/web3.js";
+import {assert} from "chai";
+import {
+  createMintV1Instruction,
+  createDecompressV1Instruction,
+  createTransferInstruction,
+  createDelegateInstruction,
+  createRedeemInstruction,
+  createCancelRedeemInstruction,
+  createCreateTreeInstruction
+} from "../sdk/bubblegum/src/generated";
+
+import {buildTree, checkProof, Tree} from "./merkle-tree";
+import {
+  decodeMerkleRoll,
+  getMerkleRollAccountSize,
+  assertOnChainMerkleRollProperties,
+  createTransferAuthorityIx,
+} from "../sdk/gummyroll";
+import NodeWallet from "@project-serum/anchor/dist/cjs/nodewallet";
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+  Token,
+} from "@solana/spl-token";
+import {execute, logTx} from "./utils";
+import {TokenProgramVersion, Version} from "../sdk/bubblegum/src/generated";
+import {sleep} from "@metaplex-foundation/amman/dist/utils";
+import {verbose} from "sqlite3";
+import {bs58} from "@project-serum/anchor/dist/cjs/utils/bytes";
+
+// @ts-ignore
+let Bubblegum;
+// @ts-ignore
+let GummyrollProgramId;
+
+/// Converts to Uint8Array
+function bufferToArray(buffer: Buffer): number[] {
+  const nums = [];
+  for (let i = 0; i < buffer.length; i++) {
+    nums.push(buffer.at(i));
+  }
+  return nums;
+}
+
+interface TreeProof  {
+  root: string,
+  proof: AccountMeta[]
+}
+
+async function getProof(asset: PublicKey): Promise<TreeProof> {
+  let resp = await fetch("http://localhost:9090", {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify({
+      jsonrpc: "2.0", id: "stupid", method: "get_asset_proof", params: [asset.toBase58()]
```

Comment:
This is very useful. Did not know how to structure RPC calls via fetch like this